queue_job/fields.py

```diff
@@ -0,0 +1,126 @@
+# copyright 2016 Camptocamp
+# license lgpl-3.0 or later (http://www.gnu.org/licenses/lgpl.html)
+
+import json
+from datetime import date, datetime
+
+import dateutil
+import lxml
+
+from odoo import fields, models
+from odoo.tools.func import lazy
+
+
+class JobSerialized(fields.Json):
+    """Provide the storage for job fields stored as json
+
+    A base_type must be set, it must be dict, list or tuple.
+    When the field is not set, the json will be the corresponding
+    json string ("{}" or "[]").
+
+    Support for some custom types has been added to the json decoder/encoder
+    (see JobEncoder and JobDecoder).
+    """
+
+    type = "job_serialized"
+    _column_type = ("jsonb", "jsonb")
+
+    _base_type = None
+
+    # these are the default values when we convert an empty value
+    _default_json_mapping = {
+        dict: "{}",
+        list: "[]",
+        tuple: "[]",
+        models.BaseModel: lambda env: json.dumps(
+            {"_type": "odoo_recordset", "model": "base", "ids": [], "uid": env.uid}
+        ),
+    }
+
+    def __init__(self, string=fields.SENTINEL, base_type=fields.SENTINEL, **kwargs):
+        super().__init__(string=string, _base_type=base_type, **kwargs)
+
+    def _setup_attrs(self, model, name):  # pylint: disable=missing-return
+        super()._setup_attrs(model, name)
+        if self._base_type not in self._default_json_mapping:
+            raise ValueError("%s is not a supported base type" % (self._base_type))
+
+    def _base_type_default_json(self, env):
+        default_json = self._default_json_mapping.get(self._base_type)
+        if not isinstance(default_json, str):
+            default_json = default_json(env)
+        return default_json
+
+    def convert_to_column(self, value, record, values=None, validate=True):
+        return self.convert_to_cache(value, record, validate=validate)
```

Comment:
shouldn't we return a PsycopgJson?

Reply:
Hi nils, updated code, also took a chance to improve `convert_to_export` (change the encoder)